main/src/ca/uwaterloo/flix/language/errors/SafetyError.scala

```diff
@@ -142,6 +142,28 @@ object SafetyError {
     override def explain(formatter: Formatter): Option[String] = None
   }
 
+  /**
+    * An error raised to indicate that a reachable entry point has an illegal signature.
+    *
+    * @param loc the location of the defn.
+    */
+  case class IllegalEntryPointSignature(loc: SourceLocation) extends SafetyError with Recoverable {
+    def summary: String = s"Illegal entry point signature"
+
+    def message(formatter: Formatter): String = {
+      import formatter._
+      s"""${line(kind, source.name)}
+         |>> Illegal entry point signature. An entry point must take a single Unit
+         |>> argument and be pure or have the IO effect.
```

Comment:
Might be confusing since we often think of the unit argument as "no arguments"

Reply:
I added a TODO to my ticket.